core/rawdb/freezer_table.go

```diff
@@ -265,6 +281,9 @@ func (t *freezerTable) repair() error {
 	contentExp = int64(lastIndex.offset)
 
 	for contentExp != contentSize {
+		if t.readonly {
```

Comment:
Do we need this check? The index file is read only and truncation will fail anyway.